packages/plugins/typescript/resolvers/src/index.ts

```diff
@@ -106,13 +106,6 @@ export type ResolverWithResolve<TResult, TParent, TContext, TArgs> = {
   const stitchingResolverUsage = `StitchingResolver<TResult, TParent, TContext, TArgs>`;
 
   if (visitor.hasFederation()) {
-    if (visitor.config.wrapFieldDefinitions) {
-      defsToInclude.push(`export type UnwrappedObject<T> = {
-        [P in keyof T]: T[P] extends infer R | Promise<infer R> | (() => infer R2 | Promise<infer R2>)
-          ? R & R2 : T[P]
-      };`);
```

Comment:
For a given type:
```ts
export type ProductMapper = {
  id: () => Promise<string>;
};
```
This type seems to return something like `(() => Promise<string>) & string` which is unusual - How can a function intersect with a string?

Regardless, since we move from `ParentType` to `FederationType`, I don't think this is needed in the future.